README.md

```diff
@@ -0,0 +1,95 @@
+  load_balancer = "${layer0_load_balancer.myapp.id}"
+  # and any other values that myapp needs
+}
+
+# Here's what we do in order to add the auth0-proxy:
+module "auth0" {
+  source = "github.com/quintilesims/auth0-proxy//terraform"
+  layer0_environment_id = "${layer0_environment.demo.id}"
+  auth0_domain = "SOME AUTH0 DOMAIN"
+  auth0_client_id = "AUTH0 CLIENT ID"
+  auth0_client_secret = "AUTH0 CLIENT SECRET"
+  auth0_redirect_uri = "https://${module.auth0.load_balancer_url}"
+  proxy_load_balancer_url = "${layer0_load_balancer.myapp.url}"
+  session_secret = "SOME RANDOM SECRET STRING"
+  ssl_certificate = "NAME OF AN SSL CERTIFICATE"
+  docker_image_tag "SOMETHING LIKE latest"
+}
+
+output "auth0_proxy_load_balancer_url" {
+  value = "https://${module.auth0.load_balancer_url}"
+}
+```
+
+Now, all traffic should access the sensitive application by using the value of the `auth0_proxy_load_balancer_url` output.
+
+## An Explanation
+
+**Note:**
+The Auth0 Proxy is intended to work with an Auth0 client configured elsewhere that is responsible for authenticating users.
+Several of the parameters that the Auth0 Proxy module requires come from the Auth0 client.
+
+| Module Parameter | Explanation |
+|---|---|
+| `source` | The location of the terraform files for the Auth0 Proxy module. This will probably always be "github.com/quintilesims/auth0-proxy//terraform". |
+| `layer0_environment_id` | The ID of the Layer0 environment in which to deploy the Auth0 Proxy module (this should be the same environment in which the sensitive application is deployed). |
+| `auth0_domain` | The domain you will use for Auth0 authentication. |
+| `auth0_client_id` | The ID of the Auth0 client to be used for authentication. |
+| `auth0_client_secret` | The secret string of the Auth0 client to be used for authentication. |
+| `auth0_redirect_uri` | The location to which Auth0 should redirect after authentication. Unless you have a different use case, this should be the URL of the Auth0 Proxy's load balancer. **Note: this must match a URL specified in the client's allowed callback URLs.** |
+| `proxy_load_balancer_url` | The location to which authenticated traffic should be directed; in other words, the private load balancer that sits in front of the sensitive application. **NOTE: This should NOT include the protocol (i.e. "http://").** |
+| `session_secret` | Any string; this is used to validate that a request is part of the Auth0 Proxy's current session. |
+| `ssl_certificate` | The Auth0 Proxy communicates over https, so you'll need a certificate. Luckily, every Layer0 instance creates one for you - something like "l0-NAMEOFLAYER0INSTANCE-api". |
+| `docker_image_tag` | The Docker image tag of the Auth0 Proxy that you want to use. Likely, you'll want to use "latest" here. |
+
+There are a few other variables with default values that can be overridden in the Auth0 module.
+You can find them at the top of the `terraform/layer0.tf` file in this repository.
```

Comment:
[nit] would be nice to have a link here.

Reply:
Good call. ✅